### DISCUSSION TITLE
Please tell me how to update in an offline environment

### DISCUSSION THREAD

**Discussion by tjdalsry** [open-webui/open-webui#3055] [Q&A]
Please tell me how to update in an offline environment

I'm using the existing version 0.24, but I want to upgrade to the new version.

Since it's a closed network, I don't know how to upgrade.

I didn't use Docker, I just installed it in an Ubuntu environment.

**Comment by JuanjoJG-WATA** [comment]
You mean a closed network without internet access at all, or without access from the outside?

How did you installed it the first time, you cloned the repo in an USB and then copied it from it?

**Reply by tjdalsry** [reply]
In the beginning, it was an internet-based environment. Now that we have entered a closed network, the Internet is no longer available.

**Comment by gsm1258** [comment]
建议用python离线安装，可以随时更新。

**Reply by tjdalsry** [reply]
My Python3 Version 3.10..

**Reply by gsm1258** [reply]
My Python Version 3.11
pip download open-webui -d d:\pip-0.3.5
pip install --upgrade --no-index --find-links=d:\pip-0.3.5 open-webui

**Reply by gsm1258** [reply]
pip install win_unicode_console==0.5
pip install compressed_rtf=1.0.6
.........
C:\.......\AppData\Local\pip\cache\wheels find .whl files.
